src/main/webapp/pipe.js

```diff
@@ -371,6 +374,32 @@ function triggerRebuild(taskId, project, buildId) {
     });
 }
 
+function triggerBuild(taskId) {
+    Q("#rebuild-" + taskId).hide();
+
+    var before;
+    if (crumb.value != null && crumb.value != "") {
+        console.info("Crumb found and will be added to request header");
+        before = function(xhr){xhr.setRequestHeader(crumb.fieldName, crumb.value);}
+    } else {
+        console.info("Crumb not needed");
+        before = function(xhr){}
+    }
+
+    Q.ajax({
+        url: rootURL + "/job/" + taskId + '/build?delay=0sec',
```

Comment:
It will be difficult to me becouse I don't use folders. If you provide me info what should I fixed or what use case I need to cover I will try to add additional changes.